boards/xtensa/intel_s1000_crb/intel_s1000_crb.dts

```diff
@@ -34,11 +35,52 @@
 
 &spi0 {
 	status = "ok";
+	#address-cells = <1>;
+	#size-cells = <1>;
 
-	mx25um512@0 {
+	flash0: flash@0 {
 		compatible = "spi,flash";
 		label = "MX25UM512";
-		reg = <0x00>;
+		reg = <0x00 0x400000>;
 		spi-max-frequency = <2000000>;
+		erase-block-size = <0x10000>;
+		write-block-size = <1>;
+	};
+};
+
+&flash0 {
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		/* The flash is 512MB in size. Each sector is 4kB and
```

Comment:
The flash size is 512Mb (as it Mega *bits*). In MB units, it is 64MB.

Reply:
Oops. Will correct. Thanks for pointing out.